feat: implement show-config subcommand (#764)
foundry-rs/starknet-foundry: docs/src/starknet/show_config.md

# Printing Current Configuration

## Overview

Sometimes, before executing any other cast command, one may want to make sure that the right
configuration settings are being used (eg proper network or account is used).

> To see just that, a `show-config` subcommand  can be used. You can just

Reviewer: ```suggestion
To see just that, a `show-config` subcommand can be used. You can just
```

# Printing Current Configuration

## Overview

> Many times before executing any other cast command you want to make sure that you are passing the right
> configuration settings and not mixing up commands that needs to be sent to development environment to mainnet :)

Reviewer: ```suggestion
Sometimes, before executing any other cast command, one may want to make sure that the right
configuration settings are being used (eg proper network or account is used).
```

wydt? :)

Author: looks good!

> For those cases you can call `show-config` with all the same configuration related arguments, you can also just
> replace subcommand and anything after it with `show-config` and it should show all the configuration that will be
> used.

Reviewer: ```suggestion
To see just that, a `show-config` subcommand  can be used. You can just
replace any subcommand (and its parameters) with `show-config` and it will show currently used configuration.
```

> Anything configuration that you cannot see there means its not set properly or is not being detected.

Reviewer: ```suggestion

```

## Examples

### General example

```shell
$ sncast \
  --rpc_url http://127.0.0.1:5050 \
  --acount user1
  show-config

command: show-config
account: user1
account_file_path: ../account-file
chain_id: alpha-goerli
keystore: ../keystore

> rpc_url: http://127.0.0.1:5050/rpc

Reviewer: let's remember to adjust the output after changes with exclusive keystore <-> accounts_file_path items

Author: yup also added separate tests for those.

Reviewer: this still shows both keystore and accounts_file_path, which are exclusive

Author: sorry forgot to update docs.
